### DISCUSSION TITLE
Resource check seems to not complete. Logs suggest a stale lock. What can we do?

### DISCUSSION THREAD

**Discussion by mrthehud** [concourse/concourse#7623] [Help & Support]
Hi all,

We have a pipeline triggered by a github-release resource. We noticed that this pipeline wasn't running, and on closer inspection, found that the github-release resource hadn't checked in over a day. The UI would show the check in progress on the resource page with a yellow border and spinner, but wouldn't show any output from it (we usually see the "selected worker" log), and would then de-highlight, and then start again.

We reviewed the logs (included below) and it appears that the lock for this resource was "already-held-locally", but there were no other jobs in progress involving this resource. There were lots of these occurring for this pipeline/resource, fairly quickly. I've included only a few.

We managed to get the pipeline running again only by destroying and recreating it - thus loosing all build history. This has happened before, and that was resolved by restarting the concourse master, I believe. Neither of these solutions seems ideal.

Does anybody have similar experience, or suggestions of configuration that may affect the locking of resources for the checks? Any suggestions on how else to resolve this state once it occurs would also be very appreciated. I ran a few searches but didn't find much that seemed to relate to this, nor any way of force unlocking a resource with fly.

I did find:
This comment for Vito, which talks about moving resource checking into the lock acquisition loop: https://github.com/concourse/concourse/issues/2010#issuecomment-409671109
This issue, about the scheduler skipping jobs, which has similar logs but we don't use time resources here: https://github.com/concourse/concourse/issues/2072

Thanks in advance,

James.

Concourse version: v7.4.0
Host OS: Linux
DB: PostgreSQL 12.7 (RDS)

This seems to be happening to several other pipelines, and has definitely started since we upgraded to 7.4.0

```json
{"timestamp":"2021-09-29T16:06:24.882444835Z","level":"debug","source":"atc","message":"atc.tracker.tick.run.get-step.lock-task.acquire.not-acquired-already-held-locally","data":{"build":"check","build_id":5821178,"id":[2,1030438438],"lock-name":"37ae469d292215a5e56d218978640b0d94aee575fbf67574347e606bf8644952","pipeline":"fvp-image-service-integration","resource":"github-release","session":"25.23273.45.5.35061.1","step-name":"github-release","team":"mds"}}
{"timestamp":"2021-09-29T16:06:24.882474748Z","level":"debug","source":"atc","message":"atc.tracker.tick.run.get-step.lock-task.did-not-get-lock","data":{"build":"check","build_id":5821178,"lock-name":"37ae469d292215a5e56d218978640b0d94aee575fbf67574347e606bf8644952","pipeline":"fvp-image-service-integration","resource":"github-release","session":"25.23273.45.5.35061","step-name":"github-release","team":"mds"}}
{"timestamp":"2021-09-29T16:06:25.721021100Z","level":"info","source":"atc","message":"atc.tracker.notify.run.check-step.initializing","data":{"build":"check","build_id":6477771,"pipeline":"fvp-image-service-integration","session":"25.43012.74.3","step-name":"github-release","team":"mds"}}
{"timestamp":"2021-09-29T16:06:26.116990237Z","level":"debug","source":"atc","message":"atc.tracker.notify.run.acquire.not-acquired-already-held-locally","data":{"build":"check","build_id":6419995,"id":[0,7494],"pipeline":"fvp-image-service-integration","resource":"github-release","session":"25.41237.2.3993","team":"mds"}}
```

**Comment by taylorsilva** [comment]
What version were you on prior to 7.4.0?
Can you share the configuration of your web nodes? Mainly interested in any feature flags you have enabled and other general web configurations. Don't care about secrets, auth, postgres settings, etc.
Are you seeing this with other types of resources or only the github-release resource?
